### PR TITLE
Advance dropdown treeview feature

### PR DESCRIPTION
**Pull request by Sina-pz**
First round of refactoring is done

Note: please update your components based on this PR to not  be affected by refactoring.

### CODE REVIEW

File: src/app/main/components/demo-page/demo-page.component.scss [hunk 1]
```diff
@@ -137,6 +137,7 @@
       padding-top: 50px;
       display: flex;
       flex-direction: column;
+      background-color: rgb(235, 235, 235);
```
Comment (dhaval2989) [inline src/app/main/components/demo-page/demo-page.component.scss:140]: use base.scss color

File: src/app/main/components/demo-page/demo-page.component.ts [hunk 2]
```diff
@@ -92,12 +93,12 @@ function mockAdvanceSearchInput(): any {
...
-      { id: 4, name: 'Anil',     age: '15', date: 'Mon Dec 200005 1995 00:00:00 GMT-0500', email: 'cohn@yahoo.com' },
-      { id: 5, name: 'Mariya',   age: '24', date: 'Mon Dec 2000005 1995 00:00:00 GMT-0500', email: 'dohn@yahoo.com' },
-      { id: 6, name: 'Crock',    age: '28', date: 'Mon Dec 200000005 1995 00:00:00 GMT-0500', email: 'eohn@yahoo.com' },
-      { id: 7, name: 'Ram',      age: '21', date: 'Mon Dec 29995 1995 00:00:00 GMT-0500', email: 'fohn@yahoo.com' },
+      { id: 2, name: 'Abraham', age: '35', date: 'Mon Dec 2005 1995 00:00:00 GMT-0500', email: 'aohn@yahoo.com' },
+      { id: 3, name: 'Sam', age: '45', date: 'Mon Dec 20005 1995 00:00:00 GMT-0500', email: 'bohn@yahoo.com' },
+      { id: 4, name: 'Anil', age: '15', date: 'Mon Dec 200005 1995 00:00:00 GMT-0500', email: 'cohn@yahoo.com' },
+      { id: 5, name: 'Mariya', age: '24', date: 'Mon Dec 2000005 1995 00:00:00 GMT-0500', email: 'dohn@yahoo.com' },
+      { id: 6, name: 'Crock', age: '28', date: 'Mon Dec 200000005 1995 00:00:00 GMT-0500', email: 'eohn@yahoo.com' },
+      { id: 7, name: 'Ram', age: '21', date: 'Mon Dec 29995 1995 00:00:00 GMT-0500', email: 'fohn@yahoo.com' },
```
Comment (dhaval2989) [inline src/app/main/components/demo-page/demo-page.component.ts:101]: create service and put hardcoded json into file.

File: src/app/main/components/demo-page/demo-page.component.html [hunk 3]
```diff
@@ -27,6 +27,12 @@
       <div class="menu-item h40" (click)="onItemClick(nvComponentType.Search)">
         {{nvComponentType[nvComponentType.Search]}}
       </div>
+      <div class="menu-item h40" (click)="onItemClick(nvComponentType.MultiSelectDropdown)">
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/demo-page/demo-page.component.html:30]: Change the component name as nv-demo-page. As we discussed, all the components should start with nv.

File: src/app/reusable/nv-checklist-dropdown/nv-checklist-dropdown.component.scss [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+    &.no-item {
+      margin-top: 10px;
+    }
+
+    .arrow-icon {
+      @include triangleArrow;
+      top: 6px;
+    }
+  }
+
```
Comment (veeranjaneyuludavuluri) [inline src/app/reusable/nv-checklist-dropdown/nv-checklist-dropdown.component.scss:76]: Same.

File: src/app/reusable/nv-checklist-dropdown/nv-checklist-dropdown.component.scss [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+}
+
+.checklist-dropdown-container {
+
```
Comment (veeranjaneyuludavuluri) [inline src/app/reusable/nv-checklist-dropdown/nv-checklist-dropdown.component.scss:80]: Same

File: src/app/reusable/nv-checklist-dropdown/nv-checklist-dropdown.component.ts [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+
+    if (index >= 0) {
+      this.items.splice(index, 1);
+    }
+    if (this.items.length === 0) {
+      this.isArrowDown = false;
+      this.hasItem = false;
+    }
+  }
+
```
Comment (veeranjaneyuludavuluri) [inline src/app/reusable/nv-checklist-dropdown/nv-checklist-dropdown.component.ts:104]: Same

File: src/app/reusable/nv-checklist-dropdown/nv-checklist-dropdown.component.ts [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+  /** Toggle a leaf to-do item selection. Check all the parents to see if they changed */
+  todoLeafItemSelectionToggle(node: TodoItemFlatNode): void {
+    this.checklistSelection.toggle(node);
+    this.checkAllParentsSelection(node);
+
+    //  save the children items to be used as chips
+    this.filteredItems = this._toChips(this.checklistSelection.selected);
+
```
Comment (veeranjaneyuludavuluri) [inline src/app/reusable/nv-checklist-dropdown/nv-checklist-dropdown.component.ts:181]: Same

File: src/assets/icons/circular-cancel--hover.icon.svg [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<svg id="ico.item.deleted" xmlns="http://www.w3.org/2000/svg" width="25.996" height="25.999" viewBox="0 0 25.996 25.999">
```
Comment (veeranjaneyuludavuluri) [inline src/assets/icons/circular-cancel--hover.icon.svg:1]: Keep as circular-cancel-hover.icon.svg

File: src/app/reusable/nv-date-picker/nv-date-picker.component.scss [hunk 9]
```diff
@@ -195,17 +149,15 @@
...
 }
 
 ::ng-deep .mat-calendar-body-cell.mat-calendar-body-in-range:after {
   background-color: $color-light-blue-10 !important;
-} 
+}
 
 ::ng-deep .mat-datepicker-content {
   box-shadow: 0px 1px 4px #0000001A !important;
   border: 1px solid #E4E4E5 !important;
```
Comment (veeranjaneyuludavuluri) [inline src/app/reusable/nv-date-picker/nv-date-picker.component.scss:162]: Use base.scss

File: src/app/reusable/nv-date-picker/nv-date-picker.component.scss [hunk 10]
```diff
@@ -195,17 +149,15 @@
...
   background-color: transparent !important;
 }
 
 ::ng-deep .mat-calendar-body-cell.mat-calendar-body-in-range:after {
   background-color: $color-light-blue-10 !important;
-} 
+}
 
 ::ng-deep .mat-datepicker-content {
   box-shadow: 0px 1px 4px #0000001A !important;
```
Comment (veeranjaneyuludavuluri) [inline src/app/reusable/nv-date-picker/nv-date-picker.component.scss:161]: Use base.scss

**Review by veeranjaneyuludavuluri — COMMENTED** [review]
Seen you didn't use prettier plugin, extra lines and indentation not correct.

**Comment by Sina-pz** [comment]
> Seen you didn't use prettier plugin, extra lines and indentation not correct.

I already have it installed; again followed your instruction about "Open setting json"

**Pull request merged**